apps/signing/views.py

```diff
@@ -4,43 +4,72 @@
 from rest_framework.response import Response
 from tower import ugettext as _
 
-from api.jwt_auth.views import JWTProtectedView
 from addons.models import Addon
+from api.jwt_auth.views import JWTProtectedView
 from devhub.views import handle_upload
+from files.models import FileUpload
 from files.utils import parse_addon
+from versions.models import Version
+from signing.serializers import FileUploadSerializer
 
 
-class UploadAddonView(JWTProtectedView):
-
-    def put(self, request, guid, version):
+def with_addon(fn):
+    def inner(view, request, guid, version):
         try:
             addon = Addon.unfiltered.get(guid=guid)
         except Addon.DoesNotExist:
-            return Response(status=status.HTTP_404_NOT_FOUND)
+            return Response({'error': _('Could not find addon.')},
+                            status=status.HTTP_404_NOT_FOUND)
         if not addon.has_author(request.user):
-            return Response(status=status.HTTP_403_FORBIDDEN)
+            return Response(
+                {'error': _('You do not own this addon.')},
+                status=status.HTTP_403_FORBIDDEN)
+        return fn(view, request, addon, version)
+    return inner
 
-        try:
+
+class UploadVersionView(JWTProtectedView):
+
+    @with_addon
+    def put(self, request, addon, version):
+        if 'upload' in request.FILES:
             filedata = request.FILES['upload']
-        except KeyError:
-            return Response({"error": "No 'upload' file found."},
+        else:
+            return Response({'error': _('No "upload" file found.')},
```

Comment:
I localised them because the `parse_addon` code also [localises them](https://github.com/mozilla/olympia/blob/02acf2283c2cb218382f0d00ad8340fb28083678/apps/files/utils.py#L445).